generate_dlc_image_release_information.py

```
@@ -0,0 +1,123 @@
+import argparse
+import logging
+import json
+import os
+import sys
+import shutil
+import tarfile
+
+import boto3
+
+from botocore.exceptions import ClientError
+from release.dlc_release_information import DLCReleaseInformation
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.addHandler(logging.StreamHandler(sys.stdout))
+LOGGER.setLevel(logging.INFO)
+
+
+def write_to_file(file_name, content):
+    """
+    Uses json package to dump content to a file. The content may or may-not be json-structured.
+    :param file_name:
+    :param content:
+    :return:
+    """
+    with open(file_name, "w") as fp:
+        fp.write(content)
+        LOGGER.info(f"Content written to file: {file_name}")
+
+
+def upload_to_S3(local_file_path, bucket_name, bucket_key):
+    """
+    Upload a local file to s3 with specified bucket name and key
+    :param local_file_path: string
+    :param bucket_name: string
+    :param bucket_key: string
+    :return:
+    """
+    _s3 = boto3.client('s3')
+    try:
+        _s3.upload_file(local_file_path, bucket_name, bucket_key)
+    except ClientError as e:
+        LOGGER.error("Error: Cannot upload file to s3 bucket.")
+        LOGGER.error("Exception: {}".format(e))
+        raise
```

Do we want an Exception to be raised here, or do we want a silent failure?

---

Wouldn't this cause the release to not proceed any further, though?